tests/api.v2/events/handleWorkRequest.test.js

```diff
@@ -1,154 +1,39 @@
-    };
-
-    try {
-      await handleWorkRequest(workRequest);
-    } catch (e) {
-      expect(e.message).toMatch(
-        /^Error: type does not match the pattern/,
-      );
-    }
-  });
-
-  it('Throws when an invalid distanceMetric in getEmbedding is encountered.', async () => {
-    // Initialize with an empty cache so a worker hit will be encountered.
-    CacheSingleton.createMock({});
-    expect.assertions(1);
-
-    const workRequest = {
-      ETag: '12345',
-      socketId: '6789',
-      experimentId: 'my-experiment',
-      timeout: '2099-01-01T00:00:00Z',
-      body: { name: 'GetEmbedding', type: 'umap', config: { minimumDistance: 0.3, distanceMetric: 'invalidMetric' } },
-    };
+const fake = require('../../test-utils/constants');
+
+jest.mock('../../../src/api.v2/middlewares/authMiddlewares');
+jest.mock('aws-xray-sdk');
+jest.mock('../../../src/api.v2/events/validateAndSubmitWork', () => (data) => {
+  if (data.uuid === 'invalid') {
+    throw new Error('invalid request');
+  }
+  return jest.fn();
+});
 
-    try {
-      await handleWorkRequest(workRequest);
-    } catch (e) {
-      expect(e.message).toMatch(
-        /^Error: distanceMetric must be one of the following:/,
-      );
-    }
-  });
 
-  it('Throws when an invalid minimumDistance in getEmbedding is encountered.', async () => {
-    // Initialize with an empty cache so a worker hit will be encountered.
-    CacheSingleton.createMock({});
-    expect.assertions(1);
+const data = {
+  uuid: 'someuuid-asd-asd-asdddasa',
+  Authorization: 'Bearer some-token',
+  experimentId: fake.EXPERIMENT_ID,
+};
 
-    const workRequest = {
-      ETag: '12345',
-      socketId: '6789',
-      experimentId: 'my-experiment',
-      timeout: '2099-01-01T00:00:00Z',
-      body: { name: 'GetEmbedding', type: 'umap', config: { minimumDistance: -1, distanceMetric: 'euclidean' } },
-    };
+describe('Handle work socket callback', () => {
+  const mockSocket = { emit: jest.fn() };
+  const mockXraySegment = { addError: jest.fn() };
 
-    try {
-      await handleWorkRequest(workRequest);
-    } catch (e) {
-      expect(e.message).toMatch(
-        /^Error: minimumDistance must be at least 0/,
-      );
-    }
+  beforeEach(() => {
+    jest.clearAllMocks();
   });
 
-  it('Throws when there is a missing config property.', async () => {
-    // Initialize with an empty cache so a worker hit will be encountered.
-    CacheSingleton.createMock({});
-    expect.assertions(1);
-
-    const workRequest = {
-      ETag: '12345',
-      socketId: '6789',
-      experimentId: 'my-experiment',
-      timeout: '2099-01-01T00:00:00Z',
-      body: { name: 'GetEmbedding', type: 'umap', config: { distanceMetric: 'euclidean' } },
-    };
-
-    try {
-      await handleWorkRequest(workRequest);
-    } catch (e) {
-      expect(e.message).toMatch(
-        /^Error: minimumDistance is a required field/,
-      );
-    }
+  it('submits work', async () => {
```

Comment:
I think you should check that the arguments passed to `validateAndSubmitWorkData` is correct.